src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkSpec.groovy

```diff
@@ -651,6 +654,50 @@ class LinkSpec extends HealthCheckSpecification {
         getIsl().srcSwitch.dpId | getIsl().srcPort | getIsl().dstSwitch.dpId | null      | "dst_port"
     }
 
+    @Tidy
+    @Tags(HARDWARE)
+    def "An ISL between noviflow switches doesn't go down when one switch is DEACTIVATED"(){
+        given: "A noviflow switch"
+        def swToDeactivate = topology.activeSwitches.find { it.noviflow }
+
+        when: "Deactivate the src switch"
+        def relatedIsl = topology.getRelatedIsls(swToDeactivate)
+        def nonNoviflowIsl = relatedIsl.find { !it.dstSwitch.noviflow }
+
+        lockKeeper.knockoutSwitch(swToDeactivate)
+        def IsSwDeactivated = true
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getSwitch(swToDeactivate.dpId).state == SwitchChangeType.DEACTIVATED
+        }
+
+        then: "All ISLs are still DISCOVERED because the system uses round trip latency for ISL alive confirmation"
+        sleep((discoveryTimeout + WAIT_OFFSET / 2) * 1000 as long)
+        // make sure that ISLs are not in FAILED status after timeout
+        Wrappers.timedLoop(WAIT_OFFSET / 2) {
+            withPool {
+                (relatedIsl - nonNoviflowIsl).everyParallel { link ->
+                    assert northbound.getLink(link).state == IslChangeType.DISCOVERED
+                }
+            }
+        }
+
+        and: "All ISLs where at least one switch is not Noviflow are FAILED"
```

Comment:
This is a bit misleading. You can have a situation when you pick a switch that do not have non-noviflow neighbors. In this case your test will state that it checked that `All ISLs where at least one switch is not Noviflow are FAILED`, but in fact it will skip this check

Reply:
fixed in the `given` block